### PR TITLE
UpdateFees Error Handling

### LINKED ISSUE #224: Appropriate error type for unable to saveAddressBook after Fees update
Currently it is throwing a warning in the constructor of LogicManager which I think violates some code quality stuff. It should also be made known to users that if the addressbook is unable to save the data upon fees update they should have a choice to exit the application and retain the previous lastUpdated.

### PR DESCRIPTION
**Pull request by angkoonhwee**
Fix #224 
**Originally**
`FeesCalculator#updateAllLessonOutstandingFeesField()` is called in the `Logic` constructor and it throws an error there if something goes wrong with the `saveAddressBook()`.
Apparently it is not a good practice to throw errors in constructors hence the change.

**Currently**
Update fees now happen at initialisation of the application. In `MainApp#initModel` , after retrieving and converting data from storage to model, `FeesCalculator` will update the model. If something goes wrong, old version of TAB with the old `lastUpdatedDate` would be used.

### CODE REVIEW

File: src/main/java/seedu/address/MainApp.java [hunk 1]
```diff
@@ -90,7 +92,19 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
...
+        // update fees upon initialisation
+        Model model = new ModelManager(initialData, userPrefs);
+        FeesCalculator feesCalculator = new FeesCalculator(model.getLastUpdatedDate(), LocalDateTime.now());
+        model = feesCalculator.updateAllLessonOutstandingFees(model);
+
+        try {
+            storage.saveAddressBook(model.getAddressBook());
+        } catch (IOException io) {
+            logger.warning("Problem from saving TAB's last updated date. The data will not be updated"
+                    + "Please restart TAB.");
```
Comment (eeliana) [inline src/main/java/seedu/address/MainApp.java:104]:
```suggestion
            logger.warning("Encountered an error while saving TAB's last updated date. The data will not be updated."
                    + " Please restart TAB.");
```

Comment (eeliana) [reply src/main/java/seedu/address/MainApp.java:104]: When will user encounter this error?

Comment (Chesterwongz) [reply src/main/java/seedu/address/MainApp.java:104]: There shouldn't be a message to restart TAB in the log I think

Comment (angkoonhwee) [reply src/main/java/seedu/address/MainApp.java:104]: when there is something wrong when saving address book, no way to replicate this also it is just a possible error that might happen

Comment (angkoonhwee) [reply src/main/java/seedu/address/MainApp.java:104]: There is no where else to put this error though @Chesterwongz

**Review by Xiaoyunnn — APPROVED** [review]
LGTM

**Review by Chesterwongz — APPROVED** [review]
LGTM

**Pull request merged**